File: LayoutTests/http/tests/security/contentSecurityPolicy/resources/save-report-and-redirect-to-save-report.php
<?php
require_once "report-file-path.php";

$DO_NOT_CLEAR_COOKIES = true; // Used by save-report.php
require_once "save-report.php";

header("HTTP/1.1 307");
header("Location: save-report.php" . (isset($_SERVER["QUERY_STRING"]) ? "?" . $_SERVER["QUERY_STRING"] : ""));
?>

File: LayoutTests/http/tests/security/contentSecurityPolicy/resources/save-report-and-redirect-to-save-report.py
#!/usr/bin/env python3

import os
import sys
sys.stdout.reconfigure(newline="")  # prevent windows \n -> \n\r conversion
from save_report import save_report

query_string = os.environ.get('QUERY_STRING', '')
if query_string != '':
    query_string = '?{}'.format(query_string)

sys.stdout.write(
    'status: 307\r\n'
    'Location: save-report.py{}\r\n'.format(query_string)
)

save_report(True)
